src/fsharp/CompileOps.fs

```diff
@@ -3520,34 +3520,37 @@ let PostParseModuleSpecs (defaultNamespace, filename, isLastCompiland, ParsedSig
 
     ParsedInput.SigFile(ParsedSigFileInput(filename, qualName, scopedPragmas, hashDirectives, specs))
 
+type ModuleNamesDict = Map<string,Map<string,QualifiedNameOfFile>>
+
 /// Checks if a module name is already given and deduplicates the name if needed.
-let DeduplicateModuleName (moduleNamesDict:IDictionary<string, Set<string>>) (paths: Set<string>) path (qualifiedNameOfFile: QualifiedNameOfFile) =
-    let count = if paths.Contains path then paths.Count else paths.Count + 1
-    moduleNamesDict.[qualifiedNameOfFile.Text] <- Set.add path paths
-    let id = qualifiedNameOfFile.Id
-    if count = 1 then qualifiedNameOfFile else QualifiedNameOfFile(Ident(id.idText + "___" + count.ToString(), id.idRange))
+let DeduplicateModuleName (moduleNamesDict:ModuleNamesDict) fileName (qualNameOfFile: QualifiedNameOfFile) =
+    let path = Path.GetDirectoryName fileName
+    let path = if FileSystem.IsPathRootedShim path then try FileSystem.GetFullPathShim path with _ -> path else path
```

Comment:
How often is `DedeplicateModuleName` called? I worry that this line could have a similar effect as #5932

Reply:
Once per check of a file - i.e. very rarely (In #5932 the GetFullPath was called like every identifier and every token in a file)

Reply:
Sounds good